Source/WebKit/UIProcess/linux/MemoryPressureMonitor.cpp

```diff
@@ -48,8 +48,13 @@ static const Seconds s_minPollingInterval { 1_s };
 static const Seconds s_maxPollingInterval { 5_s };
 static const double s_minUsedMemoryPercentageForPolling = 50;
 static const double s_maxUsedMemoryPercentageForPolling = 85;
+#if PLATFORM(WPE)
+static const int s_memoryPresurePercentageThreshold = 80;
+static const int s_memoryPresurePercentageThresholdCritical = 85;
```

Comment:
Have you tried playing with the maxPollingInterval and minPollinInterval values? Cause the poll interval is calculated a as a value between those depending on the memory that's being used (more memory used, less interval). In a fast memory usage increase situation like the one you're trying to fix, these you may have an interval bigger than what's interesting. I'd try setting both intervals to 1s. The good thing is that the this poll happens in a thread in the UI process so it doesn't hurt the performance of the WebProcess.

Reply:
Yes, I did experiment with other values. Unfortunately, the memory limits will be reached if thresholds are kept at 90% and 95% respectively even with max poll interval at 1 second.

Reply:
I think I haven't explained myself properly.
I'm trying to use the MemoryPressureMonitor to handle your situation of a fast memory increase. For that we need to be sure to detect as fast as possible the increase in memory usage. The values of maxPollInterval and minPollInterval are used to determine the interval at which the available memory is checked. The interval will be a value between those.

If there's a situation where the memory increases a lot quite fast, we can get to a situation where the poll takes several seconds to run. We don't want to happen, and that's why I recommend setting both values to 1s. This way the poll happens every 1 second always. This happens in the UI process and in its own thread, so I won't bother the performance. The thresholds that you're setting are fine, I wasn't suggesting to change them.

Then, on the WebProcess side, we will receive the memoryPressureEvents from this poll each second. The situation that we want to fix is the critical one, which will trigger synchronous memory releases, which is what we want.
The main drawback here is that handling critical requests every second is quite expensive and it's going to hurt the performance a lot while the situation persists... but it's that or crashing, so I guess it's ok.